src/crypto_tools/constants.rs

```diff
@@ -19,7 +19,8 @@ pub const PAILLIER_KEY_PROOF_TAG: u8 = 0x0B;
 pub const MODULUS_MAX_SIZE: usize = 2048;
 
 /// The min size of each prime is 1023 bits.
-pub const MODULUS_MIN_SIZE: usize = 2046;
+/// So, the product can be 2045 bits at a minimum.
+pub const MODULUS_MIN_SIZE: usize = 2045;
```

Comment:
@ggutoski Sneaking in this fix. Our tests don't catch this because the generated unsafe primes are always 1024 bits in length (whereas safe primes have that optimization).
I think this is more reason to enable some long running (non-blocking) tests that also run on safe primes.

Reply:
I realized that `1023` bits take values in the range `[2^1022, 2^1023 - 1]`, so the minimum product is `2^2044` which is `2045` bits long.
In general, multiplying `b` bit numbers gives `2b - 1` bits at a minimum.
Thinking of `2 x 2 = 4` helped me out :p